src/status_im/contexts/profile/settings/events.cljs

```diff
@@ -91,6 +92,15 @@
              (boolean private?)
              {})))
 
+(re-frame/reg-event-fx :profile.settings/toggle-peer-syncing
+ (fn [{:keys [db]}]
+   (let [value     (get-in db [:profile/profile :peer-syncing-enabled?])
+         new-value (not value)]
+     {:json-rpc/call [{:method   "wakuext_togglePeerSyncing"
+                       :params   [new-value]
+                       :on-error #(log/error "failed to toggle peer syncing" new-value %)}]
+      :db            (assoc-in db [:profile/profile :peer-syncing-enabled?] new-value)})))
+
```

Comment:
Should we add a test for this event handler?

Reply:
we recently added contract tests for testing status-go endpoints. Would be nice if you could add a test for this here @alwx

Reply:
see:https://github.com/status-im/status-mobile/pull/18665

and for more context: https://www.notion.so/Contract-Test-d47d85b444a94776998767133ab143d7